lib/active_model/serializer/reflection.rb

```diff
@@ -106,6 +106,10 @@ def value(serializer, include_slice)
       #
       def build_association(parent_serializer, parent_serializer_options, include_slice = {})
         reflection_options = options.dup
+
+        # Pass the parent's namespace onto the child serializer
+        reflection_options[:namespace] ||= parent_serializer_options[:namespace]
```

Comment:
nah, without coping the namespace from the parent to the reflection_options, the namespace isn't passed to all the children